src/redux/missions/missions.js

```diff
@@ -0,0 +1,53 @@
+import getMissionsFromAPI from '../../services/missionsAPI';
+
+const SET_MISSIONS = 'space-travelers-hub/missions/setMissions';
+const JOIN_MISSION = 'space-travelers-hub/missions/joinMission';
+const LEAVE_MISSION = 'space-travelers-hub/missions/leaveMission';
+
+const initialState = {
+  missions: [],
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case SET_MISSIONS:
+      return { ...state, missions: action.payload };
+    case JOIN_MISSION:
+      return {
+        ...state,
+        missions: state.missions.map((mission) => {
+          if (mission.mission_id === action.id) return { ...mission, reserved: true };
+          return mission;
```

Comment:
- [  ] OPTIONAL: One line `if` syntax can help you reduce the size of such simple logic operations!!

Reply:
Great suggestion! I also implemented this one. Thank you very much.